### DISCUSSION TITLE
Filtering Data Provider Using Condition With "More Than" or "Less Than" Operator

### DISCUSSION THREAD

**Discussion by chadgriff** [Budibase/budibase#6375] [Help]
I am attempting to filter a data provider based on filters defined within a Condition drawer. I am using a filter that contains the `More Than` operator. When the screen loads within the designer, I am getting an error that read `RequestError: Validation failed for parameter p3. Value must be between -2147483648 and 2147483647, inclusive`.  The value that I am trying to compare is a simple integer value that will only ever be 10, 20, 30, 40, 50, 60, or 70.

I have tested the `Less Than` operator and it results in the same error. `Equal` is working just fine.

My conditional filtering is configured around 3 different custom role id's as well as the ADMIN role (so I can see everything). 

![image](https://user-images.githubusercontent.com/15015476/174397390-114d93a8-f285-4b7a-9318-224cde465223.png)

![image](https://user-images.githubusercontent.com/15015476/174397540-08809e08-87fa-444c-8293-017e35327e89.png)

![image](https://user-images.githubusercontent.com/15015476/174397595-2e3e1a44-c4ea-4a52-84f1-16ec7667b15e.png)

I am running self hosted on Docker-Compose. I updated today and this app is at v.1.0.206.

**Comment by melohagan** [comment]
Hi @chadgriff 

What data source are you using? 
What is the data type of **statusId** in your data source?

**Comment by chadgriff** [comment]
I am using MS SQL external datasource. The statusId field is an "int" datatype in SQL and shows as a Number datatype in the fetched table within BB.

-Chad G

Sent from my iPhone

On Jun 20, 2022, at 5:10 AM, melohagan ***@***.***> wrote:

﻿

Hi @chadgriff<https://github.com/chadgriff>

What data source are you using?
What is the data type of statusId in your data source?

**Reply by melohagan** [reply]
Thank you - I was able to recreate the bug and I have raised an issue here: https://github.com/Budibase/budibase/issues/6398
